client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -206,6 +207,25 @@ public void resetReaderGroup(ReaderGroupConfig config) {
         synchronizer.updateStateUnconditionally(new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
     }
 
+    @Override
+    public Map<String, Integer> getReaderSegmentDistribution() {
+        synchronizer.fetchUpdates();
+        // fetch current state and populate assigned and unassigned distribution from the state.
+        ReaderGroupState state = synchronizer.getState();
+        ImmutableMap.Builder<String, Integer> mapBuilder = ImmutableMap.builder();
+
+        state.getOnlineReaders().forEach(reader -> {
+            Map<SegmentWithRange, Long> assigned = state.getAssignedSegments(reader);
+            int size = assigned != null ? assigned.size() : 0;
+            mapBuilder.put(reader, size);
+        });
+
+        // add unassigned against empty string
+        int unassigned = state.getNumberOfUnassignedSegments();
+        mapBuilder.put("", unassigned);
+        return mapBuilder.build();
```

Comment:
Could we add a log statement which logs the state of the ReaderGroup's state at the point of computation of the segment distribution?